src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/CABI.java

```diff
@@ -25,6 +25,8 @@
  */
 package jdk.internal.foreign;
 
+import static jdk.incubator.foreign.MemoryLayouts.ADDRESS;
+
 public enum CABI {
```

Comment:
I think if you add an explicit UNSUPPORTED constant here, a lot of the downstream code will become more linear

Reply:
Note really sure how this would help... It would mean having to add a default/UNSUPPORTED cases to switch expressions that switch over this enum. That's why I went with `null`.

Reply:
Right - all clients which switch over current ABI will have to explicitly handle case where ABI is unsupported. Right now this is done by adding a null check before the switch, which is IMHO error-prone.

Reply:
The null check is only done once inside CABI::current though, not at every `switch`. Doesn't seem that bad? (i.e. the `null` is encapsulated in CABI)

Reply:
I'll push a commit that tries this out.

Reply:
After trying this out, I agree it does look nicer, since it's clear from a caller that an exception can occur. The throwing code can be shared in SharedUtils any ways.